pubsub/pulsar/pulsar.go

```diff
@@ -202,23 +225,32 @@ func (p *Pulsar) Publish(ctx context.Context, req *pubsub.PublishRequest) error
 	)
 	topic := p.formatTopic(req.Topic)
 	producer, ok := p.cache.Get(topic)
+
+	schemaMetadata, hasSchema := p.metadata.topicSchemas[req.Topic]
```

Comment:
I'll fix this but making sure I'm seeing straight - how can this shadow a decleration in another function/scope?

@berndverst @ItalyPaleAle

Reply:
That's because it shadows the package global type definition `schemaMetadata`

Reply:
Resolved